### ISSUE TITLE
EMOTIC Baseline CNN model question

### ISSUE THREAD

**Issue by lexiwzx** [rkosti/emotic#4]
Does the EMOTIC Baseline CNN model provided mean the model with inputs B+I and trained with L_comb loss in the paper?

I've tried the model, but the performance is quite low... I'm not sure what the problem is.

**Comment by rkosti** [comment]
What is the performance you are getting? 

-Regards
Ronak

**Comment by lexiwzx** [comment]
The average AP for discrete 26 classes is 18.81. The VAD mean error rates are 0.94. 0.12, and 1.01 respectively.

> What is the performance you are getting?
> […](#)
> -Regards Ronak
> On 21 May 2019, at 11:54, Weixin Li ***@***.***> wrote: Does the EMOTIC Baseline CNN model provided mean the model with inputs B+I and trained with L_comb loss in the paper? I've tried the model, but the performance is quite low... — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by rkosti** [comment]
Did you calculate the AP using the function provided in the downloads, for fair comparison?

-Regards
Ronak

**Comment by lexiwzx** [comment]
Many thanks for your reply.

I didn't find that function in the github page. I used the average_precision_score function from Scipy. Should this be the problem? Or could you refer me to the function you mention so I can try? What are the expected AP scores based on the provided model? 


> Did you calculate the AP using the function provided in the downloads, for fair comparison?
> […](#)
> -Regards Ronak
> On 22 May 2019, at 04:10, Weixin Li ***@***.***> wrote: The average AP for discrete 26 classes is 18.81. The VAD mean error rates are 0.94. 0.12, and 1.01 respectively. What is the performance you are getting? … -Regards Ronak On 21 May 2019, at 11:54, Weixin Li @.***> wrote: Does the EMOTIC Baseline CNN model provided mean the model with inputs B+I and trained with L_comb loss in the paper? I've tried the model, but the performance is quite low... — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by rkosti** [comment]
Our model testing is based on all the images. So, we find the predictions of the model for all the test-set examples. Then use these predictions to calculate the average precision (AP). 
The dataset on our website is bigger than that in our CVPR publication. 
So, I've made changes to the repo, please use the information there to modify the way you comupute the APs.

**Comment by Sampson-Lee** [comment]
@lexiwzx Hi, have you found the solution of your problem? I got a low performance when I reproduce the experiments in PyTorch. I need some help. Thanks.

**Comment by rkosti** [comment]
The link to download the code that demonstrates how we compute the AP is here (also mentioned in the main repo as `Demo for computing the Average Precision (AP)`):  
https://www.dropbox.com/s/7bff0ucw2monn38/demoComputeAP.zip?dl=0
You should use this to compare with the performance mentioned in the paper.
@lexiwzx @Sampson-Lee
